pkg/core/blockchain.go

```diff
@@ -1577,11 +1577,46 @@ func (bc *Blockchain) NetworkFee(t *transaction.Transaction) util.Fixed8 {
 
 // SystemFee returns system fee.
 func (bc *Blockchain) SystemFee(t *transaction.Transaction) util.Fixed8 {
-	if t.Type == transaction.InvocationType {
+	switch t.Type {
+	case transaction.InvocationType:
```

Comment:
If `Version` is 0, `Gas` will be 0 (overriding any config settings).
https://github.com/neo-project/neo/blob/master-2.x/neo/Network/P2P/Payloads/InvocationTransaction.cs#L31